### PR TITLE
leaderelection: retrofit controller-manager with leaderelection client

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

**Comment by mikedanese** [comment]
@k8s-bot unit test this please

File: docs/admin/kube-controller-manager.md [hunk 1]
```diff
@@ -74,6 +74,10 @@ kube-controller-manager
       --kube-api-burst=30: Burst to use while talking with kubernetes apiserver
       --kube-api-qps=20: QPS to use while talking with kubernetes apiserver
       --kubeconfig="": Path to kubeconfig file with authorization and master location information.
+      --leader-elect[=false]: Start a leader election client and gain leadership before executing scheduler loop. Enable this when running replicated schedulers.
```
Comment (davidopp) [inline docs/admin/kube-controller-manager.md]: s/scheduler/controller manager/
s/schedulers/controller managers/

File: docs/admin/kube-controller-manager.md [hunk 2]
```diff
@@ -74,6 +74,10 @@ kube-controller-manager
...
+      --leader-elect-lease-duration=15s: The duration that non-leader candidates will wait after observing a leadershiprenewal until attempting to acquire leadership of a led but unrenewed leader slot. This is effectively the maximum duration that a leader can be stopped before it is replaced by another candidate. This is only applicable if leader election is enabled.
```
Comment (davidopp) [inline docs/admin/kube-controller-manager.md]: s/ledershiprenewal/leadership renewal/

**Comment by mikedanese** [comment]
@davidopp fixed

**Comment by mikedanese** [comment]
@k8s-bot unit test this please

File: docs/admin/kube-controller-manager.md [hunk 3]
```diff
@@ -74,6 +74,10 @@ kube-controller-manager
       --kube-api-burst=30: Burst to use while talking with kubernetes apiserver
       --kube-api-qps=20: QPS to use while talking with kubernetes apiserver
       --kubeconfig="": Path to kubeconfig file with authorization and master location information.
+      --leader-elect[=false]: Start a leader election client and gain leadership before executing scheduler loop. Enable this when running replicated components for high availability.
```
Comment (davidopp) [inline docs/admin/kube-controller-manager.md]: this still says scheduler

File: pkg/client/leaderelection/leaderelection.go [hunk 4]
```diff
@@ -354,9 +354,9 @@ func (l *LeaderElectionCLIConfig) BindFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&l.LeaderElect, "leader-elect", l.LeaderElect, ""+
 		"Start a leader election client and gain leadership before "+
 		"executing scheduler loop. Enable this when running replicated "+
```
Comment (davidopp) [inline pkg/client/leaderelection/leaderelection.go]: this still says scheduler

Comment (mikedanese) [inline pkg/client/leaderelection/leaderelection.go]: Oops sorry. Missed one. I think I got them all this time

**Comment by davidopp** [comment]
LGTM but needs rebase.

**Comment by davidopp** [comment]
@k8s-bot test this please

**Comment by mikedanese** [comment]
@k8s-bot unit test this please. Integration test flake:

```
15:36:38 F0118 23:36:38.687266    1319 controller.go:84] Unable to perform initial IP allocation check: unable to persist the updated service IP allocations: serviceipallocations "" cannot be updated: the provided resource version does not match
15:38:40 !!! [0118 23:38:40] Timed out waiting for apiserver:  to answer at http://127.0.0.1:8080/healthz; tried 120 waiting 1 between each
15:38:40 !!! Error in ./hack/test-update-storage-objects.sh:48
```

**Pull request merged**